src/linker/Linker.Steps/IAssemblyStep.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Mono.Cecil;
+using System.Collections.Generic;
+
+namespace Mono.Linker.Steps
+{
+	public interface IAssemblyStep
```

Comment:
Please use a better name, it's not Step as it can run repeatedly and I don't think we need to leak one argument type to concept name